pkg/healthcheck/healthcheck.go

```diff
@@ -801,6 +888,18 @@ func (hc *HealthChecker) PublicAPIClient() public.APIClient {
 	return hc.apiClient
 }
 
+func (hc *HealthChecker) checkLinkerdConfigConfigMap(shouldExist bool) error {
+	cm, err := hc.kubeAPI.CoreV1().ConfigMaps(hc.ControlPlaneNamespace).Get(k8s.ConfigConfigMapName, metav1.GetOptions{})
+	if err != nil {
+		if !shouldExist && kerrors.IsNotFound(err) {
```

Comment:
is this `shouldExist` check redundant now that `checkResources` handles it?

Reply:
I added it here to account for the `not found` error returned by the k8s API. Yes, you are right that we don't need to perform the `shouldExist` check here. I'll change it to:

```go
if err != nil && !kerrors.IsNotFound(err) {
  return err
}
```
And let `checkResources` do the `shouldExist` check.